src/test/regress/expected/coordinator_shouldhaveshards.out

```diff
@@ -348,11 +347,11 @@ NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinato
 -- we wont see the modifying cte in this query because we will use local execution and
 -- in postgres we wouldn't see this modifying cte, so it is consistent with postgres.
 WITH a AS (SELECT count(*) FROM test), b AS (INSERT INTO local VALUES (3,2) RETURNING *), c AS (INSERT INTO ref VALUES (3,2) RETURNING *), d AS (SELECT count(*) FROM ref JOIN local ON (a = x)) SELECT * FROM a, b, c, d ORDER BY x,y,a,b;
```

Comment:
Just noting for reference, both Citus and Postgres inlines CTEs `a` and `d`. That's why INSERTs happen first. The following is local tables where INSERTs happen first
```SQL
explain analyze WITH a AS (SELECT count(*) FROM test), b AS (INSERT INTO local VALUES (3,2) RETURNING *), c AS (INSERT INTO ref VALUES (3,2) RETURNING *), d AS (SELECT count(*) FROM ref JOIN local ON (a = x)) SELECT * FROM a, b, c, d ORDER BY x,y,a,b;
                                                           QUERY PLAN                                                            
---------------------------------------------------------------------------------------------------------------------------------
 Sort  (cost=813.62..813.62 rows=1 width=32) (actual time=0.117..0.117 rows=1 loops=1)
   Sort Key: b.x, b.y, c.a, c.b
   Sort Method: quicksort  Memory: 25kB
   CTE b
     ->  Insert on local local_1  (cost=0.00..0.01 rows=1 width=8) (actual time=0.023..0.024 rows=1 loops=1)
           ->  Result  (cost=0.00..0.01 rows=1 width=8) (actual time=0.001..0.001 rows=1 loops=1)
   CTE c
     ->  Insert on ref ref_1  (cost=0.00..0.01 rows=1 width=8) (actual time=0.005..0.006 rows=1 loops=1)
           ->  Result  (cost=0.00..0.01 rows=1 width=8) (actual time=0.001..0.001 rows=1 loops=1)
   ->  Nested Loop  (cost=813.48..813.59 rows=1 width=32) (actual time=0.104..0.108 rows=1 loops=1)
         ->  Nested Loop  (cost=38.25..38.33 rows=1 width=24) (actual time=0.045..0.049 rows=1 loops=1)
               ->  Nested Loop  (cost=38.25..38.30 rows=1 width=16) (actual time=0.038..0.040 rows=1 loops=1)
                     ->  Aggregate  (cost=38.25..38.26 rows=1 width=8) (actual time=0.010..0.010 rows=1 loops=1)
                           ->  Seq Scan on test  (cost=0.00..32.60 rows=2260 width=0) (actual time=0.006..0.006 rows=0 loops=1)
                     ->  CTE Scan on b  (cost=0.00..0.02 rows=1 width=8) (actual time=0.025..0.026 rows=1 loops=1)
               ->  CTE Scan on c  (cost=0.00..0.02 rows=1 width=8) (actual time=0.006..0.007 rows=1 loops=1)
         ->  Aggregate  (cost=775.23..775.24 rows=1 width=8) (actual time=0.056..0.056 rows=1 loops=1)
               ->  Merge Join  (cost=317.01..711.38 rows=25538 width=0) (actual time=0.047..0.051 rows=6 loops=1)
                     Merge Cond: (ref.a = local.x)
                     ->  Sort  (cost=158.51..164.16 rows=2260 width=4) (actual time=0.018..0.019 rows=1 loops=1)
                           Sort Key: ref.a
                           Sort Method: quicksort  Memory: 25kB
                           ->  Seq Scan on ref  (cost=0.00..32.60 rows=2260 width=4) (actual time=0.009..0.010 rows=1 loops=1)
                     ->  Sort  (cost=158.51..164.16 rows=2260 width=4) (actual time=0.025..0.026 rows=6 loops=1)
                           Sort Key: local.x
                           Sort Method: quicksort  Memory: 25kB
                           ->  Seq Scan on local  (cost=0.00..32.60 rows=2260 width=4) (actual time=0.005..0.007 rows=6 loops=1)
 Planning Time: 0.359 ms
 Execution Time: 0.368 ms
(29 rows)

Time: 4.788 ms
[local] onderkalaci@postgres:5432-94424=# 
```